src/ctap/clock.rs

```diff
@@ -0,0 +1,79 @@
+#[cfg(not(feature = "std"))]
+use alloc::fmt;
+pub use embedded_time::Clock;
+#[cfg(not(feature = "std"))]
+use libtock_drivers::result::FlexUnwrap;
+
+#[cfg(not(feature = "std"))]
+pub struct LibtockClock<const CLOCK_FREQUENCY: u32>(libtock_drivers::timer::Timer<'static>);
+#[cfg(not(feature = "std"))]
+impl<const CLOCK_FREQUENCY: u32> LibtockClock<CLOCK_FREQUENCY> {
+    pub fn new() -> Self {
+        let boxed_cb = alloc::boxed::Box::new(libtock_drivers::timer::with_callback(|_, _| {}));
+        let timer = alloc::boxed::Box::leak(boxed_cb).init().flex_unwrap();
+        Self(timer)
+    }
+}
+#[cfg(not(feature = "std"))]
+impl<const CLOCK_FREQUENCY: u32> fmt::Debug for LibtockClock<CLOCK_FREQUENCY> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("LibtockClock")
+            .field("CLOCK_FREQUENCY", &CLOCK_FREQUENCY)
+            .finish()
+    }
+}
+
+#[cfg(target_pointer_width = "32")]
```

Comment:
Another comment on the ClockInt. If our `TestClock` does not actually use any system clock, but returns plain zero, we could actually have the same ClockInt in tests as we have in production. If I don't overlook anything, that comes with a few advantages:
- less complexity in this file
- tests are closer to production
- less diff in this PR

Would we still need to define constants like this?
```
    const TIMEOUT_DURATION_MS: ClockInt = 100;
    const TIMEOUT_DURATION: Milliseconds<ClockInt> = Milliseconds(Self::TIMEOUT_DURATION_MS);
```
Or could that become a 1-liner?

Reply:
I changed it into one-liner and also tried to reduce the size of the PR by using inline parameter for test CtapInstant